rune/libenclave/check.go

```diff
@@ -84,12 +84,29 @@ func CreateEnclaveCgroupConfig(devices *[]*configs.Device, etype string) {
 
 // Determine whether the device is a Intel SGX enclave or AWS Nitro Enclaves device
 func enclaveMiscDev(device *configs.Device) (*configs.Device, error) {
-	dev, err := devices.DeviceFromPath(device.Path, "rwm")
+	var path string
+	var err error
+	if device.Type == 'l' {
+		path, err = os.Readlink(device.Path)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		path = device.Path
+	}
+
+	dev, err := devices.DeviceFromPath(path, "rwm")
 	if err != nil {
 		return nil, err
 	}
 
 	if dev.Type == 'c' && dev.Major == 10 {
+		// If device.Path(/dev/sgx/enclave) is a soft link while pointing to the
```

Comment:
Change to `if device.Path, e.g, /dev/sgx/enclave, is ....`

and s/soft link/symbol link/